### PR TITLE
Migrate machine_pool resource

### PR DESCRIPTION
**Pull request by JohnStrunk**
~Note: Builds on #274~

### CODE REVIEW

File: provider/machinepool/machine_pool_resource.go [hunk 1]
```diff
@@ -125,112 +125,108 @@ func (r *MachinePoolResource) Schema(ctx context.Context, req resource.SchemaReq
...
 				Description: "Select availability zone to create a single AZ machine pool for a multi-AZ cluster",
-				Type:        types.StringType,
 				Optional:    true,
 				Computed:    true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					tfsdk.UseStateForUnknown(),
-					ValueCannotBeChangedModifier(),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					//ValueCannotBeChangedModifier(),
```
Comment (nirarg) [inline provider/machinepool/machine_pool_resource.go:174]: Consider using "RequiresReplace" modifier, like done in first commit

Comment (JohnStrunk) [reply provider/machinepool/machine_pool_resource.go]: done. (I didn't notice :disappointed: )

File: provider/machinepool/machine_pool_resource.go [hunk 2]
```diff
@@ -240,81 +236,81 @@ func (r *MachinePoolResource) Create(ctx context.Context,
...
 			fmt.Sprintf(
-				"Can't build machine pool for cluster '%s, %s'", state.Cluster.Value, errMsg,
+				"Can't build machine pool for cluster '%s, %s'", state.Cluster.ValueString(), errMsg,
 			),
 		)
 		return
 	}
 
-	if !state.Replicas.Unknown && !state.Replicas.Null {
+	if !state.Replicas.IsUnknown() && !state.Replicas.IsNull() {
```
[the remark this answers is not in the document]
Comment (JohnStrunk) [reply provider/machinepool/machine_pool_resource.go]: Added `common.HasValue()`

**Pull request merged**